Tests/Get-FileHardLink.Tests.ps1

```diff
@@ -0,0 +1,155 @@
+
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+BeforeAll {
+    & (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-Test.ps1' -Resolve)
+
+    $script:testRoot = $null
+    $script:testNum = 0
+
+    function GivenHardlink
+    {
+        param(
+            [Parameter(Mandatory, Position=0)]
+            [String[]] $LinkPath,
+
+            [Parameter(Mandatory)]
+            [String] $ThatTargets
+        )
+
+        if( -not (Test-Path -Path $testRoot -PathType Container) )
+        {
+            Write-Error -Message ('Repository root "{0}" does not exist.' -f $testRoot) -ErrorAction $ErrorActionPreference
+            return
+        }
+
+        Push-Location -Path $testRoot
+
+        try
+        {
+            $targetPath = Join-Path -Path '.' -ChildPath ('.kitchen.{0}.yml' -f $ThatTargets)
+            if( -not (Test-Path -Path $targetPath -PathType Leaf) )
+            {
+                Write-Error -Message ('Target file "{0}" does not exist,' -f $targetPath) -ErrorAction $ErrorActionPreference
+                return
+            }
+
+            $targetPath = Resolve-Path -Path $targetPath | Select-Object -ExpandProperty 'ProviderPath'
+
+            foreach( $linkPath in $LinkPath )

```

Comment:
Same variable name for iterator and item. Rename the iterator item to `$linkPathItem`.